src/program/latex/mod.rs

```diff
@@ -0,0 +1,1600 @@
+            Parameter::Symbol(Symbol::Gamma) => r"\gamma".to_string(),
+            Parameter::Symbol(Symbol::Phi) => r"\phi".to_string(),
+            Parameter::Symbol(Symbol::Pi) => r"\pi".to_string(),
+            Parameter::Symbol(Symbol::Text(text)) => format(format_args!(r#"\text{{{text}}}"#)),
+        }
+    }
+}
+
+/// Settings contains the metadata that allows the user to customize how the
+/// circuit is rendered or use the default implementation.
+#[derive(Debug)]
+pub struct Settings {
+    /// Convert numerical constants, e.g. pi, to LaTeX form.
+    pub texify_numerical_constants: bool,
+    /// Include all qubits implicitly referenced in the Quil program.
+    pub impute_missing_qubits: bool,
+    /// Label qubit lines.
+    pub label_qubit_lines: bool,
+    /// Write controlled rotations in compact form.
+    pub abbreviate_controlled_rotations: bool,
+    /// Extend the length of open wires at the right of the diagram.
+    pub qubit_line_open_wire_length: u32,
+    /// Align measurement operations to appear at the end of the diagram.
+    pub right_align_terminal_measurements: bool,
+}
+
+impl Default for Settings {
+    /// Returns the default Settings.
+    fn default() -> Self {
+        Self {
+            /// false: pi is π.
+            texify_numerical_constants: true,
+            /// true: `CNOT 0 2` would have three qubit lines: 0, 1, 2.
+            impute_missing_qubits: false,
+            /// false: remove Lstick/Rstick from latex.
+            label_qubit_lines: true,
+            /// true: `RX(pi)` displayed as `X_{\\pi}` instead of `R_X(\\pi)`.
+            abbreviate_controlled_rotations: false,
+            /// 0: condenses the size of subdiagrams.
+            qubit_line_open_wire_length: 1,
+            /// false: include Meter in the current column.
+            right_align_terminal_measurements: true,
+        }
+    }
+}
+
+impl Settings {
+    /// Returns a label as the qubit name to the left side of the wire.
+    ///
+    ///  # Arguments
+    /// `name` - name of the qubit
+    pub fn label_qubit_lines(&self, name: u64) -> String {
+        Command::get_command(Command::Lstick(name.to_string()))
+    }
+
+    /// Adds missing qubits between the first qubit and last qubit in a
+    /// diagram's circuit. If a missing qubit is found, a new wire is created
+    /// and pushed to the diagram's circuit.
+    ///
+    ///  # Arguments
+    /// `&mut BTreeMap<u64, Box<Wire>> circuit` - the circuit of the diagram
+    ///
+    /// # Examples
+    /// ```
+    /// use quil_rs::{Program, program::latex::{Settings, Latex}};
+    /// use std::str::FromStr;
+    /// let program = Program::from_str("H 0\nCNOT 0 1").expect("");
+    /// let settings = Settings {
+    ///     impute_missing_qubits: true,
+    ///     ..Default::default()
+    /// };
+    /// program.to_latex(settings).expect("");
+    /// ```
+    pub fn impute_missing_qubits(&self, column: u32, circuit: &mut BTreeMap<u64, Box<Wire>>) {
```

Comment:
Likewise, not sure this needs to be defined in this `impl`. I think it could just be a top-level function with no `&self` parameter (doesn't look like that is used anywhere).

Reply:
(Addition to previous comment in this block) You're right, `&self` should be removed. These are just utility functions that  `Settings` can apply. Does it make more sense to take it out of the Settings block?